### PR TITLE
Define schemas to fix serverless warnings

### LINKED ISSUE #18: Compatibility Issue with Serverless 2.17.0
When I use this plugin, I got some warnings with recent `Serverless` version (2.17.0):

```
Serverless: Configuration warning:
Serverless:   at 'functions.publisherV1': unrecognized property 'allowAccess'
Serverless:   at 'functions.healthcheckV1': unrecognized property 'allowAccess'
Serverless:   at 'provider': unrecognized property 'access'
```
Can anyone fix this please?

### PR DESCRIPTION
**Pull request by AntonBazhal**
Closes #18 

There is no easy way to test schemas with unit tests. I checked locally on the existing project and everything seems to work.

### CODE REVIEW

File: add-permissions.js [hunk 1]
```diff
@@ -27,12 +109,7 @@ module.exports = class AwsAddLambdaAccountPermissions {
 
       if (functions.length !== 0) {
         if (policy) {
-          const { principals } = policy;
```
Comment (AntonBazhal) [inline add-permissions.js:30]: This is covered by the schema now

File: add-permissions.js [hunk 2]
```diff
@@ -67,14 +144,6 @@ module.exports = class AwsAddLambdaAccountPermissions {
 
         if (role) {
           [].concat(role).forEach(({ allowTagSession = false, maxSessionDuration = 3600, name, principals }) => {
-            if (!name) {
-              throw new Error(`Group "${groupName}" does not have role name configured`);
-            }
-
-            if (!principals) {
-              throw new Error(`Role "${name}" in the "${groupName}" group does not have principals configured`);
-            }
```
Comment (AntonBazhal) [inline add-permissions.js:76]: This is covered by the schema now

File: add-permissions.js [hunk 3]
```diff
@@ -137,10 +206,6 @@ module.exports = class AwsAddLambdaAccountPermissions {
     }
 
     const { groups } = access;
-    if (!groups) {
-      throw new Error('Access configuration must have groups defined');
-    }
```
Comment (AntonBazhal) [inline add-permissions.js:142]: This is covered by the schema now

File: add-permissions.js [hunk 4]
```diff
@@ -1,18 +1,100 @@
...
+  },
+  required: ['groups'],
+  additionalProperties: false,
+};
 
 module.exports = class AwsAddLambdaAccountPermissions {
   constructor(serverless, options) {
-    if (!semver.satisfies(serverless.version, '>= 1.12')) {
-      throw new Error('serverless-plugin-lambda-account-access requires serverless 1.12 or higher!');
-    }
```
Comment (AntonBazhal) [inline add-permissions.js:9]: This check feels useless at this point

**Review by rschick — APPROVED** [review]
Thanks Anton!

**Pull request merged**